### PR TITLE
feat: refactor refinance data types

### PR DESCRIPTION
**Pull request by robercano**
- Stopped hard casting the refinance parameters and instead rely on `superjson` instance rehydration
- Remove Zod validation at the endpoint entry level and instead use the data type guards
- Refactor some functions input parameters to type them correctly
- Add a function for refinance routing

### CODE REVIEW

File: sdk/sdk-common/src/orders/implementation/RefinanceParameters.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { IPosition } from '../../common/interfaces/IPosition'
+import { IPercentage } from '../../common/interfaces/IPercentage'
+
+/**
+ * @name RefinanceParameters
+ * @see IRefinanceParameters
+ */
+export class RefinanceParameters implements IRefinanceParameters {
+  readonly sourcePosition: IPosition
+  readonly targetPosition: IPosition
```
Comment (paszkowskiDamian) [inline sdk/sdk-common/src/orders/implementation/RefinanceParameters.ts:12]: We no longer need target position here as you refactored pool to contain just pair debt/collateral. But I can refactor this once we merge this PR.

File: sdk/sdk-common/src/user/IUser.ts [hunk 2]
```diff
@@ -1,6 +1,35 @@
...
+ */
+export const UserDataSchema = z.object({
+  wallet: WalletDataSchema,
+  chainInfo: ChainInfoDataSchema,
+})
+
+/**
+ * Type for the data part of the IUser interface
+ */
+export type IUserData = Readonly<z.infer<typeof UserDataSchema>>
```
Comment (paszkowskiDamian) [inline sdk/sdk-common/src/user/IUser.ts:26]: This one is really strange. `IUserData` has already wallet and chainInfo. Is this the **hack** that you mentioned to me today?

**Review by paszkowskiDamian — DISMISSED** [review]

File: sdk/sdk-common/src/orders/implementation/RefinanceParameters.ts (hunk 1 above)
Comment (robercano) [reply sdk/sdk-common/src/orders/implementation/RefinanceParameters.ts]: Changed!

File: sdk/sdk-common/src/user/IUser.ts (hunk 2 above)
Comment (robercano) [reply sdk/sdk-common/src/user/IUser.ts:26]: Discussed offline

**Pull request merged**